fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraVersioning.java

```diff
@@ -185,29 +186,40 @@ public Response addVersion(@HeaderParam(MEMENTO_DATETIME_HEADER) final String da
 
             final boolean createFromExisting = isBlank(datetimeHeader);
 
-            if (requestContentType == null && !createFromExisting) {
-                throw new ClientErrorException("Content Type is required for creating a binary memento",
-                        UNSUPPORTED_MEDIA_TYPE);
-            }
-
             try {
                 LOGGER.info("Request to add version for date '{}' for '{}'", datetimeHeader, externalPath);
 
                 // Create memento
-                final FedoraResource memento;
-                if (resource instanceof FedoraBinary) {
-                    memento = versionService.createBinaryVersion(session.getFedoraSession(), resource(),
-                            mementoInstant, null, null, null, null);
-                } else {
+                FedoraResource memento = null;
+                final boolean isBinary = resource instanceof FedoraBinary;
+                if (isBinary) {
+                    final FedoraBinary binaryResource = (FedoraBinary) resource;
```

Comment:
This is probably my lack of understanding, but if `isBinary == true` because `resource instanceof FedoraBinary` then why this line?

Reply:
need to pass a FedoraBinary object into `versionService.createBinaryVersion`, so I cast it once for passing around. Unless I'm misunderstanding the question.

Reply:
Sorry the question was around the fact that on line 193 you check if `resource instanceof FedoraBinary` and if it is, then you cast it as a `FedoraBinary`.

Reply:
`resource` is declared as a `FedoraResource`, which FedoraBinary extends, so I'm just narrowing it down to a more specific implementation.